pandas/tests/arithmetic/test_period.py

```diff
@@ -1085,3 +1085,21 @@ def test_pi_sub_period_nat(self):
         exp = pd.TimedeltaIndex([np.nan, np.nan, np.nan, np.nan], name='idx')
         tm.assert_index_equal(idx - pd.Period('NaT', freq='M'), exp)
         tm.assert_index_equal(pd.Period('NaT', freq='M') - idx, exp)
+
+
+class TestPeriodArithmetic(object):
+
+    @pytest.mark.parametrize('n', [1, 2, 3, 4])
+    @pytest.mark.parametrize('freq,expected', [
+        (pd.offsets.Second, 18489600),
+        (pd.offsets.Minute, 308160),
+        (pd.offsets.Hour, 5136),
+        (pd.offsets.Day, 214),
+        (pd.offsets.MonthEnd, 7),
+        (pd.offsets.YearEnd, 1),
+    ])
```

Comment:
definitely needs cases with kwargs passed to offset constructors.  Putting it in tests.tseries.offsets might be useful since there are test classes that construct a bunch of these

Reply:
parameterized the tests on some kwargs too, only YearEnd takes any kwds besides normalize ('month'). And only non-Tick offsets (in this case MonthEnd and YearEnd) can take normalize=True.